scripts/generate_model.py

```diff
@@ -1,170 +1,365 @@
+boxes and inertia, so spinning the propeller does affect the simulation.
 """
 
 import math
 import re
-import sys
+from argparse import ArgumentParser
+
+import yaml
 
 # SDF 1.9 supports degrees="true"; provide some nice vars for earlier versions
 d180 = math.pi
 d90 = d180 / 2
 d45 = d90 / 2
+d30 = d90 / 3
 d135 = d90 + d45
 
-mass = 10
-visual_x = 0.457
-visual_y = 0.338
-visual_z = 0.25
-fluid_density = 1000
-
-# The ROV should be positively buoyant
-buoyancy_adjustment = 0.05
-displaced_mass = mass + buoyancy_adjustment
-
-# The collision box is used by the BuoyancyPlugin
-# collision_x * collision_y * collision_z * density == displaced_mass
-collision_x = visual_x
-collision_y = visual_y
-collision_z = displaced_mass / (visual_x * visual_y * fluid_density)
-
-# The center of mass is just above the origin
-mass_z = 0.011
-
-# The center of volume is directly above the center of mass, resulting in a restoring force
-volume_z = 0.06
-
-ixx = mass / 12 * (collision_y * collision_y + collision_z * collision_z)
-iyy = mass / 12 * (collision_x * collision_x + collision_z * collision_z)
-izz = mass / 12 * (collision_x * collision_x + collision_y * collision_y)
-
-# 2nd order stability for the HydrodynamicsPlugin
-xUabsU = -0.5 * visual_y * visual_z * 0.8 * fluid_density
-yVabsV = -0.5 * visual_x * visual_z * 0.95 * fluid_density
-zWabsW = -0.5 * visual_x * visual_y * 0.95 * fluid_density
-kPabsP = -0.5 * 0.008 * fluid_density
-mQabsQ = -0.5 * 0.008 * fluid_density
-nRabsR = -0.5 * 0.008 * fluid_density
-
-# Thruster placement
-thruster_x = 0.14
-thruster_y = 0.092
-thruster_z = -0.009
-vert_thruster_y = 0.109
-vert_thruster_z = 0.077
-
-# Propeller link parameters
-propeller_size = "0.1 0.02 0.01"
-propeller_mass = 0.002
-propeller_ixx = 0.001
-propeller_iyy = 0.001
-propeller_izz = 0.001
-
-# ThrusterPlugin parameters
-propeller_diameter = 0.1
-thrust_coefficient = 0.02
-
-# Max thrust force, N
-# Both forward and reverse thrust must be the same
-max_thrust = 50
-
-# ArduPilotPlugin control parameters
-servo_min = 1100
-servo_max = 1900
-control_offset = -0.5
-
-# From the command line
-use_angvel_cmd = False
-
-# Set by update_globals()
-cw_control_multiplier = 0   # Thrusters 3, 4 and 6
-ccw_control_multiplier = 0  # Thrusters 1, 2 and 5
-thruster1_topic = "/model/bluerov2/joint/thruster1_joint/cmd_"
-thruster2_topic = "/model/bluerov2/joint/thruster2_joint/cmd_"
-thruster3_topic = "/model/bluerov2/joint/thruster3_joint/cmd_"
-thruster4_topic = "/model/bluerov2/joint/thruster4_joint/cmd_"
-thruster5_topic = "/model/bluerov2/joint/thruster5_joint/cmd_"
-thruster6_topic = "/model/bluerov2/joint/thruster6_joint/cmd_"
-
-
-# Fossen equation, see "Guidance and Control of Ocean Vehicles" p. 246
-def thrust_to_ang_vel(thrust):
+# Density of water
+fluid_density = 1000.0
+
+
+def thrust_to_ang_vel(
+    thrust: float,
+    propeller_diameter: float,
+    thrust_coefficient: float,
+    fluid_density: float = 998,
```

Comment:
This shadows the global var of the same name, and the default is different. I like the config files; why not move fluid_density into the config? This gets rid of all globals except for the helper "dxxx" vars.